### DISCUSSION TITLE
how to disable or get the commands for the docker-compose.yml formatter

### DISCUSSION THREAD

**Discussion by dantuck** [microsoft/vscode-docker#3602] [Q&A] [answered]
Can someone point me to where the docker-compose.yml formatter is? I either need to be able to disable it or get the commands ran so we can add it into a pre-commit and/or CI check flow.

Thanks in advance!

**Comment by bwateratmsft** [comment, marked as answer]
Converting this to a discussion as that seems to fit better.

The formatter is in the [Compose language service](https://github.com/microsoft/compose-language-service). You can disable it with this setting, either in your `.vscode/settings.json` or in user settings:
```json
    "[dockercompose]": {
        "editor.formatOnSave": false
    }
```

I'm not aware of any way to use VSCode language servers as a precommit hook or CI check. Ultimately this particular formatter is just parsing it in with the [`yaml` library](https://www.npmjs.com/package/yaml) and rewriting it with a certain number of spaces for indentation; one could create an executable NPM script / package to do the same.

**Reply by dantuck** [reply]
Thank you for your time answering this. I think I have a path forward now!

**Reply by bwateratmsft** [reply]
Glad to help!
